docs/metrics.md

```diff
@@ -0,0 +1,435 @@
+
+We have two **categories** of events here: rendering events and interaction events:  `lockbox_setup.render` and `lockbox_setup.interaction`. For rendering events, the **object** field is populated by the name of the view being rendered; for interaction events the **object** is the name of the button. Below is a brief description of the view names and their corresponding buttons.
+
+-   `lockbox_toolbar_firstrun`
+    -   Pop-over shown when the user first clicks on the Lockbox toolbar icon after install. Contains the `get_started_button`. Note that the click on the toolbar icon itself is logged in a separate event (see "Toolbar Button Interaction Event" section below).
+-   `reenter_password`
+    -   Shown after FxA process is completed. Prompts user to re-enter their FxA pw to start using lockbox. Contains the `lockbox_signin_button`
+
+
+The **methods** for these events are `render` and `click` for rendering and interaction events, respectively.
+
+
+The **extra** field contains the user's FxA anonymous/random user id. For `lockbox_toolbar_firstrun` view this will be `null`.
+
+
+To register the rendering events:
+```javascript
+Services.telemetry.registerEvents("lockbox_setup.render", {
+  "lockbox_setup.render": {
+    methods: ["render"],
+    objects: ["lockbox_toolbar_firstrun", // startup flow views
+            "reenter_password"],
+    extra: {"fxa_uid": uid} // value is null in some cases, see above
+  }
+});
+```
+To register the interaction events:
+```javascript
+Services.telemetry.registerEvents("lockbox_setup.interaction", {
+  "lockbox_setup.click": {
+    methods: ["click"],
+    objects: ["get_started_button", // start-up flow buttons
+              "lockbox_signin_button"],
+    extra: {"fxa_uid": uid} // value is null in some cases, see above
+  }
+});
+```
+An example of how to record a rendering of the `reenter_password` view:
+```javascript
+Services.telemetry.recordEvent("lockbox_setup.render",
+                    "render",
+                    "reenter_password",
+                    null,
+                    {"fxa_uid": uid} ) // user has signed in by now);
+```                    
+
+
+To record (for example) a click on one of the `lockboxSignin_button` button:
+
+```javascript
+Services.telemetry.recordEvent("lockbox_setup.interaction",
+                    "click",
+                    "lockbox_signin_button",
+                    null,
+                    {"fxa_uid": null} ) // user hasn't signed in yet);
+```                    
+### Toolbar Button Interaction Event
+At this stage of development, the Lockbox toolbar icon is the main entrypoint for using the Lockbox UI. There is only one event of concern here, a click on the icon. This event has its own **category** of `toolbar.interaction`. The **method** is click, and the object is **toolbar_button**.
+
+The event can be registered as follows:
+
+```javascript
+Services.telemetry.registerEvents("toolbar.interaction", {
```

Comment:
Strictly speaking, this event can only be recorded if a popup is **not** registered.

That said, we could assume this event occurred if a popup view is rendered and explicitly record a "toolbar.click" event when a popup view is rendered (in addition to or instead of other events), or infer this event if one of the popup view rendering events is recorded.

Reply:
Since we already have a render event for the item list, I'll remove this event from the doc